src/ioctls/vm.rs

```diff
@@ -880,11 +881,26 @@ impl VmFd {
         target_arch = "arm",
         target_arch = "aarch64"
     ))]
-    pub fn register_irqfd(&self, fd: &EventFd, gsi: u32) -> Result<()> {
-        let irqfd = kvm_irqfd {
-            fd: fd.as_raw_fd() as u32,
-            gsi,
-            ..Default::default()
+    pub fn register_irqfd(
+        &self,
+        fd: &EventFd,
+        resample_fd: Option<&EventFd>,
```

Comment:
I think it would make the code more readable if we would define a new method, e.g. `register_irqfd_with_resample` instead of adding an optional parameter here. Then you should also add in the `Arguments` section of that method the new parameter and some info about it.
Even if using `Option` is the preferred way here, we can still remove some code duplication by initializing a mutable default kvm_irqfd, and then mutating the resample related fields when we have a `Some`.

Reply:
Thanks for your review -
I updated the `Arguments` and added a description of how the behavior is modified if resamplefd is given.
I changed the `kvm_irqfd` to be mutable and modify when there is `Some` resamplefd.

Regarding  `Option` vs adding another function e.g. `register_irqfd_and_resamplefd`:
- Adding optional resamplefd makes this function more directly match the `KVM_IRQFD` ioctl
- It's not much harder to read/use - users of kvm-ioctls are probably familiar with the kvm api anyway
On the other hand,
- There is already a separate function to unregister `fd`. Maybe it is a bit clearer and more idiomatic to create a new function for this new case/behavior.
- Changing the function signature of `register_irqfd` adds a little work for downstream
Maybe these are good enough reason to create a new function, instead...I'll think on it some more

Reply:
Since the kvm-ioctls already does not map 1:1 with the KVM API, we could add the function as suggested by @lauralt because then we would also not break the API for existing consumers of this crate. Carrying a `None` in your function calls when you're not using resampling makes the code a bit harder to read as well because you don't immediately know what the `None` refers to.

Our goal with this crate is to make it easy to use for all users, not only the ones that are already familiar with KVM. That's why we also have extensive documentation and code examples.

Reply:
Thank you for the additional feedback!
I've updated this to be a new function `register_irqfd_with_resample`, as @lauralt originally suggested.